Item adapter diff

Tiemoue opened the pull request:
this diff allows user to view items in the closet in  a grid recycler-view

Tiemoue committed:
added closet item.xml

Tiemoue committed:
added closet item.xml

Tiemoue committed:
updated string resources

Tiemoue committed:
updated fragment_closet.xml

Tiemoue committed:
Created Adapter for the Recycler-view

Tiemoue committed:
Updated closet Fragment and deleted AddItemActivity.java

sazonovale commented on app/src/main/java/com/example/easycloset/ClosetFragment.java:
```diff
@@ -32,7 +49,39 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
+        query.addDescendingOrder("createdAt");
+        query.findInBackground((items, e) -> {
+            // check for errors
+            if (e != null) {
```
don't you want to close progress dialog if error happens? Show error toast? You can test it on you phone if you put it in flight mode

Tiemoue replied:
fixed

sazonovale commented on app/src/main/java/com/example/easycloset/ClosetFragment.java:
```diff
@@ -1,29 +1,46 @@
+    }
+
+    @Override
+    public void onResume() {
```
what is the reason you fetch in onResume? 

It does not look right to me because if this flow happens:
1. user opened ClosetFragment
2. onResume called, load started, load finished
3. user press Home button
4. user go back to ClosetFragment
???
6. fetch started again even if we still have items in cache

So point is that fetching every time is not necessary

Tiemoue replied:
fixed

Tiemoue replied:
removed queryPost from onResume and use onActivityResult to check if an item has removed then queryPost

Tiemoue committed:
Addressed Feedback

Tiemoue committed:
addressed feedback

Tiemoue committed:
addressed feedback

Tiemoue committed:
addressed feedback

sazonovale commented on app/src/main/java/com/example/easycloset/ClosetFragment.java:
```diff
@@ -32,7 +39,40 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
+        queryPosts();
+    }
+
+    public ItemsAdapter getAdapter() {
```
is there a reason why this is public?

Tiemoue replied:
no. I've changed them to private

Tiemoue committed:
addressed feedback:
changed public methods to private, removed queryPost from onResume and into onViewCreated and close App if theres an error in query post.
